sound/soc/sof/sof-client-dma-trace.c

```diff
@@ -0,0 +1,691 @@
+// SPDX-License-Identifier: GPL-2.0-only
+//
+// Copyright(c) 2018-2021 Intel Corporation. All rights reserved.
+//
+// Author: Liam Girdwood <liam.r.girdwood@linux.intel.com>
+//
+// SOF client version:
+//  Peter Ujfalusi <peter.ujfalusi@linux.intel.com>
+//
+
+#include <asm/unaligned.h>
+#include <linux/debugfs.h>
+#include <linux/module.h>
+#include <linux/pm_runtime.h>
+#include <linux/slab.h>
+#include <linux/sched/signal.h>
+#include <sound/sof/header.h>
+#include <sound/sof/info.h>
+#include <sound/sof/trace.h>
+#include <sound/pcm.h>
+#include "sof-client.h"
+#include "sof-client-dma-trace.h"
+#include "sof-utils.h"
+
+/* dma-trace platform implementation type */
+enum sof_dtrace_platform {
+	SOF_DTRACE_PLATFORM_SOF = 0,
+	SOF_DTRACE_PLATFORM_HOST_ASSISTED,
```

Comment:
What do the definitions refer to? what's the different between 'SOF' and 'HOST_ASSISTED'.

Reply:
@plbossart, in earlier versions I had different strings as match, like 'hda-trace', atom-trace', imx-trace', etc. This does not made much sense to continue when AMD and Mediatek is also going to need to add a string.
So, one step back and we have two types of dma-trace implementation:
- solely confined within the firmware (atom, imx, etc)
- dtrace which needs assistance from the host side to work (HDA and AMD)

`SOF_DTRACE_PLATFORM_SOF` ("snd_sof.sof-dma-trace") is for platforms which does not need host side ops.
`SOF_DTRACE_PLATFORM_HOST_ASSISTED` ("snd_sof.host-assisted-dma-trace") is for platforms _needing_ host assistance, host ops.

Reply:
OK